jetson/science_bridge/src/__main__.py

```diff
@@ -249,22 +248,44 @@ def mosfet_transmit(self, channel, msg):
         pass
 
     def nav_status(self, channel, msg):  # TODO - fix make better
-        print("Received nav req")
         # Off, Done, Else
-
         struct = NavStatus.decode(msg)
+
+        if self.previous_auton_msg == struct.nav_state_name:
+            return
+
+        print("Received new nav req")
+
         message = "$Mosfet,{device},{enable}"
 
-        # Off = Blue
+        blue_off = message.format(device=Mosfet_devices.BLUE_LED.value, enable=0)
+        blue_off = self.add_padding(blue_off)
+        self.ser.write(bytes(blue_off, encoding='utf-8'))
+        red_off = message.format(device=Mosfet_devices.RED_LED.value, enable=0)
+        red_off = self.add_padding(red_off)
+        self.ser.write(bytes(red_off, encoding='utf-8'))
+        green_off = message.format(device=Mosfet_devices.GREEN_LED.value, enable=0)
+        green_off = self.add_padding(green_off)
+        self.ser.write(bytes(green_off, encoding='utf-8'))
+
+        self.previous_auton_msg = struct.nav_state_name
+
         if struct.nav_state_name == "Off":
-            print("navstatus off")
+            prev = Auton_state.OFF
+        elif struct.nav_state_name == "Done":
+            prev = Auton_state.DONE
+        else:
+            prev = Auton_state.ELSE
```

Comment:
maybe instead Auton_state.ON? ELSE fits the logic but the general idea is ELSE means auton is on (but not done)

Reply:
done